### PR TITLE
Use libblockdev swap plugin for swapspace

### PR DESCRIPTION
**Pull request by vojtechtrefny**
New version of #124

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -483,6 +483,12 @@ AC_SUBST([PART_LDFLAGS])
 CFLAGS=$SAVE_CFLAGS
 LDFLAGS=$SAVE_LDFLAGS
 
+# libblockdev swap
+SWAP_CFLAGS="-I/usr/include/blockdev"
+SWAP_LDFLAGS="-lbd_swap"
+AC_SUBST([SWAP_CFLAGS])
+AC_SUBST([SWAP_LDFLAGS])
+
```
Comment (tsmetana) [inline configure.ac]: If I'm correct the libblockdev libraries/headers may be installed independently. I know the rest of the configure script contains similar things all over but I would really prefer AC_CHECK_*() or PKG_CHECK_MODULES() so the compilation fails early enough when the dependecies are missing. Moreover there is also a SPEC file template in packaging/, its BuildRequires should be updated too.

File: src/udiskslinuxswapspace.c [hunk 2]
```diff
@@ -244,22 +260,41 @@ swapspace_stop_on_job_completed (UDisksJob   *job,
...
+               GError            **error)
+{
+
+  UDisksObject *object = NULL;
+  UDisksBlock *block = NULL;
+  gchar *device = NULL;
+  gboolean ret = FALSE;
+
+  object = UDISKS_OBJECT (g_dbus_interface_get_object (G_DBUS_INTERFACE (user_data)));
+  block = udisks_object_peek_block (object);
```
Comment (vpodzime) [inline src/udiskslinuxswapspace.c]: From the docs:
```
It is not safe to use the returned object if you are on another thread than the one where the GDBusObjectManagerClient or GDBusObjectManagerServer for object is running.
```
``udisks_object_get_block ()`` should be used instead as that increases the ref counter.

File: src/udiskslinuxswapspace.c [hunk 3]
```diff
@@ -244,22 +260,41 @@ swapspace_stop_on_job_completed (UDisksJob   *job,
...
+  device = g_strdup (udisks_block_get_device (block));
```
Comment (vpodzime) [inline src/udiskslinuxswapspace.c]: Again from the docs:
```
The returned value is only valid until the property changes so on the client-side it is only safe to use this function on the thread where object was constructed. Use udisks_block_dup_device() if on another thread.
```

File: src/udiskslinuxswapspace.c (hunk 2 above)
Comment (vojtechtrefny) [inline src/udiskslinuxswapspace.c]: Fixed.

File: src/udiskslinuxswapspace.c (hunk 3 above)
Comment (vojtechtrefny) [inline src/udiskslinuxswapspace.c]: Fixed.

File: configure.ac (hunk 1 above)
Comment (vojtechtrefny) [inline configure.ac]: I hope I've fixed this one too :)

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, this is ok to test.

**Comment by vpodzime** [comment]
The tests failure is unrelated to changes in this PR.

**Pull request merged**